CHANGELOG.md

```diff
@@ -5,6 +5,7 @@
 ### Enhancements
 
 - Implemented database optimization routine (#721).
+- Improve checks during account update in the store (#739). 
```

Comment:
I think this is actually a bugfix - or is there no way to trigger this in production code? Possibly the `block-producer` or prover would disallow this? cc @bobbinth 

We should mark this as a `### Fix` maybe.

```suggestion
- Harden store account updates (#739). 
```